src/io/inputs/data-series-csv-importer/DataSeriesRepoImporter.cpp

```
@@ -19,116 +19,158 @@
+            col[rowIndex] = val;
+            columns.emplace_back(std::move(col));
+            ptr = next;
+            ++colIndex;
         }
-        std::vector<double> row = readLine(line, csvSeparators, path.filename().string());
-        if (!result.empty() && row.size() != result[0].size())
+        else
         {
-            throw std::invalid_argument(path.filename().string()
-                                        + ": rows have inconsistent number of columns");
+            if (colIndex < columns.size())
+            {
+                columns[colIndex][rowIndex] = val;
+                ptr = next;
+                ++colIndex;
+            }
+            else
+            {
+                throw std::invalid_argument(errorMessagePrefix
+                                            + ": rows have inconsistent number of columns");
+            }
         }
-        result.push_back(row);
     }
-    return result;
+    if (rowIndex != 0 && colIndex != columns.size())
+    {
+        throw std::invalid_argument(errorMessagePrefix
+                                    + ": rows have inconsistent number of columns");
+    }
+    return colIndex != 0;
 }
 
-static TimeSeriesSet matrixToTimeSeriesSet(std::string id, std::vector<std::vector<double>> matrix)
+static std::vector<std::vector<double>> readCSV(const std::filesystem::path& filename, char sep)
 {
-    // We have to transpose the matrix
-    // TODO: we may want to improve this by reading directly into the TimeSeriesSet object, or
-    // by creating a specific IDataSeries implementation
-    int nTimesteps = matrix.size();
-    TimeSeriesSet timeSeriesSet(id, nTimesteps);
-    if (nTimesteps == 0)
+    // MappedFile file(fileName.c_str());
+    // Check file size first
+    std::error_code ec;
+    auto sz = std::filesystem::file_size(filename, ec);
+    if (ec || sz == 0)
     {
-        return timeSeriesSet;
+        return {}; // empty or inaccessible
     }
```

IMO we should also throw an exception here, but display a more specific error message

the `return {}` is dangerous, it could lead to invisible errors.

My suggestion 
```cpp
if (sz == 0) return {}; // empty file -> empty matrix
if (ec) throw std::runtime_error("Error reading file " + filename + "(" + ec.message() +")");
```